src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/StorageDataReader.cs

```diff
@@ -93,6 +93,15 @@ public Task<bool> ReadAsync(CancellationToken cancellationToken)
             return DbDataReader.ReadAsync(cancellationToken);
         }
 
+        /// <summary>
+        /// Pass-through to DbDataReader.Read()
+        /// </summary>
+        /// <returns></returns>
+        public bool Read()
```

Comment:
Would suggest adding a big warning to ReadAsync with information about the performance issues (and same as below, links to an issue/docs would be preferred)

Reply:
I'm gonna delete it eventually - need to move it away from EditData area as well - planned for another PR.